src/pybind/rbd/setup.py

```diff
@@ -148,11 +149,15 @@ def check_sanity():
     sys.exit(1)
 
 cmdclass = {}
+compiler_directives={'language_level': sys.version_info.major}
 try:
     from Cython.Build import cythonize
     from Cython.Distutils import build_ext
+    from Cython import __version__ as cython_version
 
     cmdclass = {'build_ext': build_ext}
+    if version.parse(cython_version) >= version.parse('3'):
```

Comment:
Need to document this:
```suggestion
    
    # Needed for building with Cython 0.x and Cython 3 from the same file,
    # preserving the same behavior.
    # When Cython 0.x builds go away, replace this compiler directive with
    # noexcept on rbd_callback_t and librbd_progress_fn_t (or consider doing
    # something similar to except? -9000 on rbd_diff_iterate2() callback for
    # progress callbacks to propagate exceptions).
    if version.parse(cython_version) >= version.parse('3'):
```